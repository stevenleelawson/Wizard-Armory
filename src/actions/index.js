export const loadCards = (cards, color) => ({
  type: 'LOAD_CARDS',
  cards,
  color
});

export const formState = (color) => ({
  type: 'FORM_STATE',
  color
});

export const addToDeck = (card) => ({
  type: 'ADD_TO_DECK',
  card
});

export const clearDeck = () => ({
  type: 'CLEAR_DECK'
})
