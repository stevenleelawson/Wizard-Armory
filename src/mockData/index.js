export const mockCardData = [{
  name: "As Luck Would Have It",
  manaCost: "{G}",
  cmc: 1,
  type: "Enchantment",
  rarity: "Rare",
  set: "UST",
  setName: "Unstable",
  text: "Hexproof Whenever you roll a die, put a number of luck counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
  artist: "Milivoj Ä†eran",
  number: "102",
  layout: "normal",
  multiverseid: 439491,
  imageUrl: "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card",
  originalText: "Hexproof Whenever you roll a die, put a number of luck, counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
  originalType: "Enchantment",
  id: "8d2722825dfa475154e0c87ba72c47528c4cf485"
}];
