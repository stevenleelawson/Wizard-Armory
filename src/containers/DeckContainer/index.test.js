import React from 'react';
import { shallow } from 'enzyme';
import { DeckContainer, mapDispatchToProps, mapStateToProps } from './index';
import * as mockData from '../../mockData';
import * as actions from '../../actions';

describe('DeckContainer', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<DeckContainer deck={mockData.mockCardData}/>);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('mapDispatchToProps', () => {
  it('should call dispatch with the right parmas for addToDeck', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    const expected = actions.addToDeck(mockData.mockCleanCard);
    mapped.addToDeck(mockData.mockCleanCard);
    expect(mockDispatch).toHaveBeenCalledWith(expected);
  });
});

describe('mapStateToProps', () => {

  it('should return an array with cards equal to an array', () => {
    const expected = { deck: mockData.mockCleanCard};
    const actual = mapStateToProps(expected);
    expect(actual).toEqual(expected);
  });

  it('should return an object with cards equal to an array', () => {
    const expected = { cards: mockData.mockCleanCard};
    const actual = mapStateToProps(expected);
    expect(actual).toEqual(expected);
  });
});
