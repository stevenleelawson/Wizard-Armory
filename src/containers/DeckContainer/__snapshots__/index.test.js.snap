// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckContainer should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    deck={
        Array [
            Object {
              "artist": "Milivoj Ćeran",
              "cmc": 1,
              "id": "8d2722825dfa475154e0c87ba72c47528c4cf485",
              "imageUrl": "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card",
              "layout": "normal",
              "manaCost": "{G}",
              "multiverseid": 439491,
              "name": "As Luck Would Have It",
              "number": "102",
              "originalText": "Hexproof Whenever you roll a die, put a number of luck, counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
              "originalType": "Enchantment",
              "rarity": "Rare",
              "set": "UST",
              "setName": "Unstable",
              "text": "Hexproof Whenever you roll a die, put a number of luck counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
              "type": "Enchantment",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          addToDeck={undefined}
          deck={
                    Array [
                              Object {
                                "artist": "Milivoj Ćeran",
                                "cmc": 1,
                                "id": "8d2722825dfa475154e0c87ba72c47528c4cf485",
                                "imageUrl": "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card",
                                "layout": "normal",
                                "manaCost": "{G}",
                                "multiverseid": 439491,
                                "name": "As Luck Would Have It",
                                "number": "102",
                                "originalText": "Hexproof Whenever you roll a die, put a number of luck, counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
                                "originalType": "Enchantment",
                                "rarity": "Rare",
                                "set": "UST",
                                "setName": "Unstable",
                                "text": "Hexproof Whenever you roll a die, put a number of luck counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
                                "type": "Enchantment",
                              },
                            ]
          }
          id="8d2722825dfa475154e0c87ba72c47528c4cf485"
          imageUrl="http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card"
          info={
                    Object {
                              "artist": "Milivoj Ćeran",
                              "cmc": 1,
                              "id": "8d2722825dfa475154e0c87ba72c47528c4cf485",
                              "imageUrl": "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card",
                              "layout": "normal",
                              "manaCost": "{G}",
                              "multiverseid": 439491,
                              "name": "As Luck Would Have It",
                              "number": "102",
                              "originalText": "Hexproof Whenever you roll a die, put a number of luck, counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
                              "originalType": "Enchantment",
                              "rarity": "Rare",
                              "set": "UST",
                              "setName": "Unstable",
                              "text": "Hexproof Whenever you roll a die, put a number of luck counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
                              "type": "Enchantment",
                            }
          }
/>,
      ],
      "className": "deck-card-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "8d2722825dfa475154e0c87ba72c47528c4cf485",
        "nodeType": "function",
        "props": Object {
          "addToDeck": undefined,
          "deck": Array [
            Object {
              "artist": "Milivoj Ćeran",
              "cmc": 1,
              "id": "8d2722825dfa475154e0c87ba72c47528c4cf485",
              "imageUrl": "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card",
              "layout": "normal",
              "manaCost": "{G}",
              "multiverseid": 439491,
              "name": "As Luck Would Have It",
              "number": "102",
              "originalText": "Hexproof Whenever you roll a die, put a number of luck, counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
              "originalType": "Enchantment",
              "rarity": "Rare",
              "set": "UST",
              "setName": "Unstable",
              "text": "Hexproof Whenever you roll a die, put a number of luck counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
              "type": "Enchantment",
            },
          ],
          "id": "8d2722825dfa475154e0c87ba72c47528c4cf485",
          "imageUrl": "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card",
          "info": Object {
            "artist": "Milivoj Ćeran",
            "cmc": 1,
            "id": "8d2722825dfa475154e0c87ba72c47528c4cf485",
            "imageUrl": "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card",
            "layout": "normal",
            "manaCost": "{G}",
            "multiverseid": 439491,
            "name": "As Luck Would Have It",
            "number": "102",
            "originalText": "Hexproof Whenever you roll a die, put a number of luck, counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
            "originalType": "Enchantment",
            "rarity": "Rare",
            "set": "UST",
            "setName": "Unstable",
            "text": "Hexproof Whenever you roll a die, put a number of luck counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
            "type": "Enchantment",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            addToDeck={undefined}
            deck={
                        Array [
                                    Object {
                                      "artist": "Milivoj Ćeran",
                                      "cmc": 1,
                                      "id": "8d2722825dfa475154e0c87ba72c47528c4cf485",
                                      "imageUrl": "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card",
                                      "layout": "normal",
                                      "manaCost": "{G}",
                                      "multiverseid": 439491,
                                      "name": "As Luck Would Have It",
                                      "number": "102",
                                      "originalText": "Hexproof Whenever you roll a die, put a number of luck, counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
                                      "originalType": "Enchantment",
                                      "rarity": "Rare",
                                      "set": "UST",
                                      "setName": "Unstable",
                                      "text": "Hexproof Whenever you roll a die, put a number of luck counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
                                      "type": "Enchantment",
                                    },
                                  ]
            }
            id="8d2722825dfa475154e0c87ba72c47528c4cf485"
            imageUrl="http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card"
            info={
                        Object {
                                    "artist": "Milivoj Ćeran",
                                    "cmc": 1,
                                    "id": "8d2722825dfa475154e0c87ba72c47528c4cf485",
                                    "imageUrl": "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card",
                                    "layout": "normal",
                                    "manaCost": "{G}",
                                    "multiverseid": 439491,
                                    "name": "As Luck Would Have It",
                                    "number": "102",
                                    "originalText": "Hexproof Whenever you roll a die, put a number of luck, counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
                                    "originalType": "Enchantment",
                                    "rarity": "Rare",
                                    "set": "UST",
                                    "setName": "Unstable",
                                    "text": "Hexproof Whenever you roll a die, put a number of luck counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
                                    "type": "Enchantment",
                                  }
            }
/>,
        ],
        "className": "deck-card-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "8d2722825dfa475154e0c87ba72c47528c4cf485",
          "nodeType": "function",
          "props": Object {
            "addToDeck": undefined,
            "deck": Array [
              Object {
                "artist": "Milivoj Ćeran",
                "cmc": 1,
                "id": "8d2722825dfa475154e0c87ba72c47528c4cf485",
                "imageUrl": "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card",
                "layout": "normal",
                "manaCost": "{G}",
                "multiverseid": 439491,
                "name": "As Luck Would Have It",
                "number": "102",
                "originalText": "Hexproof Whenever you roll a die, put a number of luck, counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
                "originalType": "Enchantment",
                "rarity": "Rare",
                "set": "UST",
                "setName": "Unstable",
                "text": "Hexproof Whenever you roll a die, put a number of luck counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
                "type": "Enchantment",
              },
            ],
            "id": "8d2722825dfa475154e0c87ba72c47528c4cf485",
            "imageUrl": "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card",
            "info": Object {
              "artist": "Milivoj Ćeran",
              "cmc": 1,
              "id": "8d2722825dfa475154e0c87ba72c47528c4cf485",
              "imageUrl": "http://gatherer.wizards.com/Handlers/Image.ashx?multiverseid=439491&type=card",
              "layout": "normal",
              "manaCost": "{G}",
              "multiverseid": 439491,
              "name": "As Luck Would Have It",
              "number": "102",
              "originalText": "Hexproof Whenever you roll a die, put a number of luck, counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
              "originalType": "Enchantment",
              "rarity": "Rare",
              "set": "UST",
              "setName": "Unstable",
              "text": "Hexproof Whenever you roll a die, put a number of luck counters on As Luck Would Have It equal to the result. Then if there are 100 or more luck counters on As Luck Would Have It, you win the game. (Count both rolls if you reroll a die.)",
              "type": "Enchantment",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
